internal/assert/assert.go

```diff
@@ -17,54 +17,28 @@ package assert
 import (
 	"testing"
 
-	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
-	"github.com/golang/protobuf/proto"
-	"github.com/golang/protobuf/ptypes"
-	"github.com/golang/protobuf/ptypes/any"
 	"github.com/google/go-cmp/cmp"
-	"github.com/google/go-cmp/cmp/cmpopts"
+	tassert "github.com/stretchr/testify/assert"
+	"google.golang.org/protobuf/testing/protocmp"
 )
 
-type Assert struct {
-	t *testing.T
-}
+var (
+	Equal         = tassert.Equal
+	Equalf        = tassert.Equalf
+	ElementsMatch = tassert.ElementsMatch
+)
 
-// Equal will test that want == got, and call t.Fatal if it does not.
-// Notably, for errors, they are equal if they are both nil, or are both non-nil.
-// No value information is checked for errors.
-func Equal(t *testing.T, want, got interface{}, opts ...cmp.Option) {
+// EqualProto will test that want == got for protobufs, call t.Error if it does not,
+// and return a bool to indicate the result. This mimics the behavior of the testify `assert`
+// functions.
+func EqualProto(t *testing.T, want, got interface{}, msgAndArgs ...interface{}) bool {
 	t.Helper()
-	Assert{t}.Equal(want, got, opts...)
-}
 
-// Equal will call t.Fatal if want and got are not equal.
-func (a Assert) Equal(want, got interface{}, opts ...cmp.Option) {
```

Comment:
I know there are a couple places where we were previously passing in `opts`, and I don't see them being removed. Looks like since `tassert.Equal` also takes a variadic arg of type `interface{}` those occurrences won't cause compile errors, but it'd still be good to clean them up since they are no longer relevant. I'd start by looking for `cmp.AllowUnexported` (e.g. https://github.com/projectcontour/contour/blob/master/internal/dag/policy_test.go#L170).

Reply:
Good catch, thanks!

Reply:
I made a temporary copy of `Equal` without the variadic params, then fixed up all those tests. Testify `assert` doesn't need the AllowUnexported thing, so all the tests work. I broke a few to check. :)